frontend/src/hooks/useApi.ts

```diff
@@ -0,0 +1,65 @@
+import { useState, useCallback } from 'react';
+import { API } from 'aws-amplify';
+
+const baseHeaders: HeadersInit = {
+  'Content-Type': 'application/json',
+  Accept: 'application/json'
+};
+
+type ApiMethod = (apiName: string, path: string, init?: any) => Promise<any>;
+type OnError = (e: Error) => Promise<void>;
+
+export const useApi = (onError?: OnError) => {
+  const [requestCount, setRequestCount] = useState(0);
+
+  const getToken = () => {
+    const t = localStorage.getItem('token');
+    return t ? JSON.parse(t) : '';
+  };
+
+  const prepareInit = useCallback(async (init: any) => {
+    const { headers, ...rest } = init;
+    return {
+      ...rest,
+      headers: {
+        ...headers,
+        ...baseHeaders,
+        Authorization: getToken()
+      }
+    };
+  }, []);
+
+  const apiMethod = useCallback(
+    (method: ApiMethod) => async <T extends object = any>(
+      path: string,
+      init: any = {}
+    ) => {
+      console.log('REQUEST', path);
+      try {
+        setRequestCount(cnt => cnt + 1);
+        const options = await prepareInit(init);
+        const result = await method('crossfeed', path, options);
+        setRequestCount(cnt => cnt - 1);
+        return result as T;
+      } catch (e) {
+        setRequestCount(cnt => cnt - 1);
+        onError && onError(e);
+        throw e;
+      }
+    },
+    [prepareInit, onError]
+  );
+
+  const api = {
+    apiGet: useCallback(apiMethod(API.get.bind(API)), [apiMethod]),
```

Comment:
Why do we need to use `.bind(API)` here?

Reply:
Bind `API` as the `this` argument when `API.get` is called. Happens automatically when calling `API.get()`, but passing as an argument requires this.